pkgs/games/funkin/default.nix

```diff
@@ -0,0 +1,141 @@
+{ stdenv, lib, fetchFromGitHub, fetchpatch
+, haxe, haxePackages, neko, makeWrapper
+, alsaLib, libpulseaudio, libGL, libX11, libXdmcp, libXext, libXi, libXinerama, libXrandr
+, makeDesktopItem
+}:
+
+let
+  desktopItem = makeDesktopItem {
+    name = "funkin";
+    exec = "Funkin";
+    desktopName = "Friday Night Funkin";
+    categories = ["Game" "ArcadeGame"];
+    icon = "funkin";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "funkin";
+  version = "unstable-2021-04-02";
+
+  src = fetchFromGitHub {
+    owner = "ninjamuffin99";
+    repo = "Funkin";
+    rev = "faaf064e37d5a150d8aba451d740eeb81bd2e974";
+    sha256 = "053qbmib9nsgx63d0rcykky6284cixibrwq9rv1sqx2ghsdn78ff";
+  };
+
+  patches = [
+    # TODO remove when fix pushed
+    # Fixes Freeplay crash on master
+    (fetchpatch {
+      url = "https://github.com/ninjamuffin99/Funkin/pull/412/commits/af62bdca4d70e06ea8b383ab5c9c6b3f5d8f087b.patch";
+      sha256 = "1p9vzm7hkay8fzrd82hgrnafd0g5qi34brzv0hyjirryhni840fy";
+    })
+    # Fixes bugged version outdated version
+    (fetchpatch {
+      name = "fix-the-update-check.patch";
+      url = "https://github.com/ninjamuffin99/Funkin/pull/941/commits/996c9b6d89bf4447b1f39390bb6e8fb7d6a8c37c.patch";
+      sha256 = "1m5b47y2pbdwdk5dsvgivq8705cz95n8imdckyg3r9jazhkrvh6k";
+    })
+    # Fix link opening
+    (fetchpatch {
+      name = "Makes-the-donation-button-actually-work-on-Linux.patch";
+      url = "https://github.com/ninjamuffin99/Funkin/pull/172.patch";
+      sha256 = "1491fw52j5am5bg02v05y82njbdj1aqqdidxz36i4g6kcdy2x2iz";
+    })
+  ];
+
+  postPatch = ''
+    # Real API keys are stripped from repo
+    cat >source/APIStuff.hx <<EOF
+    package;
+
+    class APIStuff
+    {
+      public static var API:String = "";
+      public static var EncKey:String = "";
+    }
+    EOF
+  '';
+
+  nativeBuildInputs = [ haxe neko makeWrapper ]
+  ++ (with haxePackages; [
+    hxcpp
+    hscript
+    openfl
+    lime
+    flixel
+    flixel-addons
+    flixel-ui
+    newgrounds
+    polymod
+    discord_rpc
+  ]);
+
+  buildInputs = [ alsaLib libpulseaudio libGL libX11 libXdmcp libXext libXi libXinerama libXrandr ];
+
+  enableParallelBuilding = true;
+
+  buildPhase = ''
+    runHook preBuild
+
+    export HOME=$PWD
+    haxelib run lime build linux -final
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/{bin,lib/funkin}
+    cp -R export/release/linux/bin/* $out/lib/funkin/
+    for so in $out/lib/funkin/{Funkin,lime.ndll}; do
+      $STRIP -s $so
+    done
+    wrapProgram $out/lib/funkin/Funkin \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath buildInputs} \
+      --run "cd $out/lib/funkin"
+    ln -s $out/{lib/funkin,bin}/Funkin
+
+    # desktop file
+    mkdir -p $out/share/applications
+    ln -s ${desktopItem}/share/applications/* $out/share/applications
+
+    # icons
+    mkdir -p $out/share/icons/hicolor/16x16/apps
```

Comment:
maybe use a loop

https://grep.app/search?q=16%2032&regexp=true&filter[repo][0]=NixOS/nixpkgs&filter[lang][0]=Nix&filter[path][0]=pkgs/

Reply:
Nice website ! (and so, escaping ${x} is done by ''${x} ? I didn't knew that and got me in some trouble in the past). Done